src/main/java/com/github/davidmoten/rx/jdbc/State.java

```diff
@@ -6,8 +6,8 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 class State {
-    volatile Connection con;
-    volatile PreparedStatement ps;
-    volatile ResultSet rs;
+    Connection con;
```

Comment:
You're probably right. I would like to have an unit test for this case, but couldn't get one ready after a quick hacking. I will try to finish this later next week.